packages/nouns-contracts/contracts/NounsAuctionHouseV2.sol

```diff
@@ -0,0 +1,428 @@
+        bool success;
+        assembly {
+            success := call(30000, to, value, 0, 0, 0, 0)
+        }
+        return success;
+    }
+
+    /**
+     * @notice Set historic prices; only callable by the owner, which in Nouns is the treasury (timelock) contract.
+     * @dev This function lowers auction price accuracy from 18 decimals to 10 decimals, as part of the price history
+     * bit packing, to save gas.
+     * @param settlements The list of historic prices to set.
+     */
+    function setPrices(Settlement[] memory settlements) external onlyOwner {
+        uint256[] memory nounIds = new uint256[](settlements.length);
+        uint256[] memory prices_ = new uint256[](settlements.length);
+
+        for (uint256 i = 0; i < settlements.length; ++i) {
+            settlementHistory[settlements[i].nounId] = SettlementState({
+                blockTimestamp: settlements[i].blockTimestamp,
+                amount: ethPriceToUint64(settlements[i].amount),
+                winner: settlements[i].winner
+            });
+
+            nounIds[i] = settlements[i].nounId;
+            prices_[i] = settlements[i].amount;
+        }
+
+        emit HistoricPricesSet(nounIds, prices_);
+    }
+
+    /**
+     * @notice Warm up the settlement state for a list of Noun IDs.
+     * @dev Helps lower the gas cost of auction settlement when storing settlement data
+     * thanks to the state slot being non-zero.
+     * @dev Only writes to slots where blockTimestamp is zero, meaning it will not overwrite existing data.
+     * @param nounIds The list of Noun IDs whose settlement slot to warm up.
+     */
+    function warmUpSettlementState(uint256[] calldata nounIds) external {
+        for (uint256 i = 0; i < nounIds.length; ++i) {
+            if (settlementHistory[nounIds[i]].blockTimestamp == 0) {
+                settlementHistory[nounIds[i]] = SettlementState({ blockTimestamp: 1, amount: 0, winner: address(0) });
+            }
+        }
+    }
+
+    /**
+     * @notice Get past auction prices.
+     * @dev Returns prices in reverse order, meaning settlements[0] will be the most recent auction price.
+     * @param auctionCount The number of price observations to get.
+     * @return settlements An array of type `Settlement`, where each Settlement includes a timestamp,
+     * the Noun ID of that auction, the winning bid amount, and the winner's addreess.
+     */
+    function prices(uint256 auctionCount) external view returns (Settlement[] memory settlements) {
+        uint256 latestNounId = auction.nounId;
+        if (!auction.settled && latestNounId > 0) {
+            latestNounId -= 1;
+        }
+
+        settlements = new Settlement[](auctionCount);
+        uint256 actualCount = 0;
+        while (actualCount < auctionCount && latestNounId > 0) {
+            // Skip Nouner reward Nouns, they have no price
+            // Also skips IDs with no price data
+            if (settlementHistory[latestNounId].blockTimestamp == 0) {
```

Comment:
@eladmallel if someone called `warmUpSettlementState`, then the `blockTimestamp` will be `== 1` and this check will be false.
so we need to check `<= 1`, or look at `winner` or `amount`
I'm wondering if we need to keep this warmUp function at all.

Reply:
fixed: 05eb47a49affc06cef39675a5d80c663ae48df03